torchvision/transforms/_geometry.py

```diff
@@ -0,0 +1,59 @@
+from typing import Any, Dict, Optional, Tuple
+
+import torch
+
+from torchvision.features import BoundingBox, Image
+
+from ._core import Transform, query_sample
+
+__all__ = ["HorizontalFlip", "RandomRotate", "RandomErase"]
+
+
+class HorizontalFlip(Transform):
+    @staticmethod
+    def image(input: Image) -> Image:
+        return Image(input.flip((-1,)))
+
+    @staticmethod
+    def bounding_box(input: BoundingBox) -> BoundingBox:
+        x, y, w, h = input.convert("xywh").to_parts()
+        x = input.image_size[1] - (x + w)
+        return BoundingBox.from_parts(x, y, w, h, like=input, format="xywh")
+
+
+class RandomRotate(Transform):
+    def __init__(self, low: float, high: float) -> None:
+        super().__init__()
+        self._dist = torch.distributions.Uniform(low, high)
+
+    def get_params(self, sample: Any = None) -> Dict[str, Any]:
+        return dict(degrees=self._dist.sample().item())
```

Comment:
Returning all the random parameters in a single function call will likely not work for iterative algorithms such as:
https://github.com/pytorch/vision/blob/96f6e0a117d5c56f7e0237851dbb96144ebb110b/references/detection/transforms.py#L81-L129

The same might apply for data augmentation techniques such as [UniformAugment](https://arxiv.org/abs/2003.14348) where there is a random number of looping through transforms that themselves have random parameters. Example:
https://github.com/automl/trivialaugment/blob/master/aug_lib.py#L672-L677

In such cases, it might be hard to have a Functional form that does not have side-effects due to calling random.

Reply:
very interesting and challenging point. So if I understand well, the only way to get around those use cases is to have a transform design that takes all objects to be transformed through the same function at the same time, as they might need to be able to 'see' each other. Am I right?

Reply:
That seems to be the case. This should be doable in the current approach, since `get_params` receives the full sample and thus has access to all the information.